js/widgets/forms/textinput.js

```diff
@@ -38,7 +38,7 @@ return $.widget( "mobile.textinput", {
 	initSelector: "input[type='text']," +
 		"input[type='search']," +
 		":jqmData(type='search')," +
-		"input[type='number']," +
+		"input[type='number']:not(:jqmData(type='range'))," +
```

Comment:
OK ... removing one ```:not()``` doesn't seem to be making much difference - probably because there's a second ```:not()``` at the end of the ```initSelector```.